lib/ret/media_search.ex

```diff
@@ -507,6 +514,21 @@ defmodule Ret.MediaSearch do
   defp add_tag_to_listing_search_query(query, nil), do: query
   defp add_tag_to_listing_search_query(query, tag), do: query |> where(fragment("tags->'tags' \\? ?", ^tag))
 
+  defp add_similar_to_to_listing_search_query(query, nil), do: query
+
+  defp add_similar_to_to_listing_search_query(query, similar_sid) do
+    case Avatar.avatar_or_avatar_listing_by_sid(similar_sid) do
```

Comment:
There is probably some way to do this without the additional query, but not sure how. That said this probably needs to eventually walk the hierarchy to find listings with any common ancestors, and I think that would need an additional query anyway.